File: code-coaches/medium/flowing-words/FlowingWords.cs
using System;

class FlowingWords
{
    static bool DoesSentenceFlow(string sentence)
    {
        string[] words = sentence.Split(' ');
        for (int i = 1; i < words.Length; ++i)
        {
            if (words[i][0] != words[i - 1][words[i - 1].Length - 1])
            {
                return false;
            }
        }
        return true;
    }

    static void Main(string[] args)
    {
        string sentence = Console.ReadLine();

        if (DoesSentenceFlow(sentence))
            Console.WriteLine("true");
        else
            Console.WriteLine("false");
    }
}

File: code-coaches/medium/flowing-words/FlowingWords.java
import java.util.Scanner;

public class FlowingWords {

    public static boolean doesSentenceFlow(String sentence) {
        String[] words = sentence.split(" ");
        for (int i = 1; i < words.length; i++) {
            if (!words[i].startsWith(words[i - 1].substring(words[i - 1].length() - 1))) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String sentence = scanner.nextLine();

        if (doesSentenceFlow(sentence)) {
            System.out.println("true");
        } else {
            System.out.println("false");
        }

        scanner.close();
    }

}
